packages/accounts-controller/src/AccountsController.test.ts

```diff
@@ -0,0 +1,1519 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import type { InternalAccount } from '@metamask/keyring-api';
+import { EthAccountType, EthMethod } from '@metamask/keyring-api';
+import type { SnapControllerState } from '@metamask/snaps-controllers';
+import { SnapStatus } from '@metamask/snaps-utils';
+import * as uuid from 'uuid';
+
+import type {
+  AccountsControllerActions,
+  AccountsControllerEvents,
+  AccountsControllerState,
+} from './AccountsController';
+import { AccountsController, keyringTypeToName } from './AccountsController';
+
+jest.mock('uuid');
+const mockUUID = jest.spyOn(uuid, 'v4');
+
+const defaultState: AccountsControllerState = {
+  internalAccounts: {
+    accounts: {},
+    selectedAccount: '',
+  },
+};
+
+const mockGetKeyringForAccount = jest.fn();
+const mockGetKeyringByType = jest.fn();
+const mockGetAccounts = jest.fn();
+
+const mockAccount: InternalAccount = {
+  id: 'mock-id',
+  address: '0x123',
+  options: {},
+  methods: [...Object.values(EthMethod)],
+  type: EthAccountType.Eoa,
+  metadata: {
+    name: 'Account 1',
+    keyring: { type: 'HD Key Tree' },
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    lastSelected: 1691565967656,
```

Comment:
Would be nice to have a comment here to explain what are we ignoring

Reply:
We can remove this. The previous version of keyring-api didn't have lastSelected yet. Changed in https://github.com/MetaMask/core/commit/aaaa3ae28ab63baa70ec94027efc8f3d97b8b003